bentoml/yatai/repository/abs_repository.py

```
@@ -0,0 +1,151 @@
+# limitations under the License.
+
+import logging
+from datetime import datetime, timedelta
+import os
+from urllib.parse import urlparse
+from dependency_injector.wiring import Provide, inject
+
+from bentoml.configuration.containers import BentoMLContainer
+from bentoml.exceptions import YataiRepositoryException
+from bentoml.yatai.proto.repository_pb2 import BentoUri
+from bentoml.yatai.repository.base_repository import BaseRepository
+
+logger = logging.getLogger(__name__)
+
+
+class ABSRepository(BaseRepository):
+    @inject
+    def __init__(
+        self,
+        base_url,
+        expiration: int = Provide[
+            BentoMLContainer.config.yatai.repository.abs.expiration
+        ],
+    ):
+        try:
+            from azure.storage.blob import (
+                BlobServiceClient,
+                generate_blob_sas,
+                AccountSasPermissions,
+            )
+        except ImportError:
+            raise YataiRepositoryException(
+                '"azure-storage-blob" package is required for Azure Blob Storage.'
+                'You can install it with pip: '
+                '"pip install pip install azure-storage-blob"'
+                'Find out more at https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python'  # noqa: E501
+            )
+        self.generate_blob_sas = generate_blob_sas
+        # fmt: off
+        self.sas_permission_add = AccountSasPermissions(
+            create=True,
+            write=True,
+        )
+        self.sas_permission_get = AccountSasPermissions(
+            read=True,
+        )
+        # fmt: on
+
+        self.uri_type = BentoUri.ABS
+        self.base_url = base_url
+
+        parsed_url = urlparse(base_url)
+        self.account = parsed_url.netloc.split(".", 1)[0]
+        self.container, self.blob = parsed_url.path.split("/", 1)
+
+        connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
```

We want to proactively throw error, if the connection string is missing